src/Compilers/CSharp/Portable/Binder/InMethodBinder.cs

```diff
@@ -36,7 +36,7 @@ public IteratorInfo(TypeSymbol elementType, ImmutableArray<Diagnostic> elementTy
         }
 
         public InMethodBinder(MethodSymbol owner, Binder enclosing)
-            : base(enclosing)
+            : base(enclosing, enclosing.Flags & ~BinderFlags.AllClearedAtInMethodBoundary)
```

Comment:
>AllClearedAtInMethodBoundary [](start = 61, length = 28)

I don't think it is necessary to introduce a new set of flags. Instead, I would assert in the body of the constructor that enclosing.Flags doesn't contain flags that we cannot create tests for.

Reply:
Alright. Just added an assert.